assets/javascripts/discourse/components/chat-message-collapser.js

```diff
@@ -40,4 +41,38 @@ export default Component.extend({
   close() {
     this.set("collapsed", true);
   },
+
+  @computed("cooked")
+  get isYoutube() {
+    return isYoutube(this.cooked);
+  },
+
+  @computed("uploads", "cooked")
+  get isImage() {
+    return isImage(this.uploads, this.cooked);
+  },
+
+  @computed("cooked", "message")
+  get isAnimatedImage() {
+    return isAnimatedImage(this.cooked, this.message);
+  },
 });
+
+function isYoutube(cooked) {
+  return /^<div class="onebox lazyYT lazyYT-container"/.test(cooked);
+}
+
+function isAnimatedImage(cooked, message) {
+  const onebox = `<p><img src="${escape(message)}" class="animated onebox"`;
+  return cooked.startsWith(onebox);
+}
+
+function isImage(uploads) {
+  return uploads?.length > 0;
```

Comment:
What happens if the uploads here are PDFs or other non-image documents?

Reply:
As of before-this-commit, they show up like that and are available for download:

<img width="311" alt="Screenshot 2022-01-10 at 10 19 17 PM" src="https://user-images.githubusercontent.com/1555215/148780901-7674edb3-682f-4378-984d-9248dd40d771.png">

Reply:
Cool! So the only tiny comment then is that `isImage` is a misnomer.

Reply:
Your comments have been addressed here https://github.com/discourse/discourse-chat/pull/504